crates/cxx-qt-gen/src/generator/utils/rust.rs

```diff
@@ -0,0 +1,79 @@
+// SPDX-FileCopyrightText: 2022 Klarälvdalens Datakonsult AB, a KDAB Group company <info@kdab.com>
+// SPDX-FileContributor: Andrew Hayzen <andrew.hayzen@kdab.com>
+//
+// SPDX-License-Identifier: MIT OR Apache-2.0
+
+use syn::{GenericArgument, PathArguments, Result, ReturnType, Type, TypeReference};
+
+/// When a [syn::ReturnType] is used in a CXX bridge we need to ensure that
+/// there is only one generic
+pub(crate) fn syn_return_type_for_cxx_bridge(ty: &ReturnType) -> Result<ReturnType> {
+    // TODO: if Result<T, E> change to Result<T>
+
+    // FIXME: problem is that we can't detect anyhow::Result<T>
```

Comment:
problem is this can't work as we can't determine the type.

The new API described in #531 is the better approach.